src/Xunit.Categories/ArchitectureTestDiscoverer.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections.Generic;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Xunit.Categories
+{
+	public class ArchitectureTestDiscoverer : ITraitDiscoverer
+	{
+		internal const string DiscovererTypeName = DiscovererUtil.AssemblyName + "." + nameof(ArchitectureTestDiscoverer);
+
+		public IEnumerable<KeyValuePair<string, string>> GetTraits(IAttributeInfo traitAttribute)
+		{
+            yield return new KeyValuePair<string, string>("Category", "ArchitectureTest");
+        }
```

Comment:
It looks like there's some mixed tabs/spaces for indentation here:

```suggestion
        {
            yield return new KeyValuePair<string, string>("Category", "ArchitectureTest");
        }
```